doc updates (#44)
statistical-perceptions/serverless-survey: docs/source/researcher/qualtrics.md

# Coordinating with Qualtrics 

## Basic use

> 1. A first qualtrics survey sends embedded data via forwarding only

Reviewer: embedded data is participantid = id, right? I feel like that needs to be explained somewhere. Participant ID is assigned by the first survey, equated to id as it is passed along via HTML to the serverless survey, then passed on as id moving forward.... Probably deserves its own section somewhere.

> 2. (optional) Middle qualtrics surveys receive data and send data 

Reviewer: ```suggestion
2. If your survey length is too long, we recommend making an intermediary qualtrics survey to receive data  from one set of questions and pass along the subject id. 
```

Reviewer: Will there be some explaination on how to merge the datasets (or recommended documentation)?

3. The final Qualtrics survey receives data only

### Sending the ID from Qualtrics

> 1. Add an embedded data block with the identifier to forward (eg panelID or Response ID) 

Reviewer: I don't understand this

> 2. Redirect end of survey to the URL of your stud

Reviewer: ```suggestion
2. Redirect end of survey to the URL of your study
```

Reviewer: What exactly does this mean? Are you directin the researcher to use the last next_question_url variable to link out to their final Qualtrics form?

3. embed the response id in the forwarding url:

Template
```

> https://<gh_repo_owner>.github.io/<repo_name>/<question_out_html_file>.html?id=<qualtrics piped text>

Reviewer: ```suggestion
https://<gh_repo_owner>.github.io/<repo_name>/<question_out_html_file>.html?id=<qualtrics piped text>
Example
 https://statistical-perceptions.github.io/fairalg/bbq1.html?id=${e://Field/ResponseID}
```

Reviewer: Could be useful to explain:

1. Where you find repo_name
2. Where you find question_out_html_file <-- this is your first question_id that you want your participant to see
3. How to generate the Qualtrics piped text field

```
where: 
- `<gh_repo_owner>` is the name of the repo owner, i.e. an organization or gh username; if you set up a [custom domain](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site) this and the `.github.io` would be replaced with that. 
- `<repo_name>` is the name of the repository; 
- `<question_out_html_file>` comes from your config file choices, for the first question participants should be directed to. 

> - `<qualtrics piped text>` is for whatever you chose as the embedded data above. 

Reviewer: Are these all things in the YML file or where are we?

::::{important}
your github org name (or custom domain) and repo name will be visible to your study participants 
:::::::

Example:

```
https://statistical-perceptions.github.io/sample-nobackend/?id=${e://Field/ResponseID}
```
in this case, `statistical-perceptions` is the owner, `sample-nobackend` is the repo name and the study only has one question page so there is no question_id set and I used the ResponseID feild as the embedded data for the identifier. 

### Recieving Data into Qualtrics

> 1. set up embedded data as the first block on the workflows tab. set the variables as per the instructions output by [getting data from url](https://www.qualtrics.com/support/survey-platform/survey-module/survey-flow/standard-elements/embedded-data/#SettingValuesFromTheSurveyURL)

Reviewer: ```suggestion
1. In Qualtrics, add an Embedded Data block at the top of the Survey Flow. This is where you define the variables that will receive data passed through the URL. Set these variables according to the instructions in [getting data from url](https://www.qualtrics.com/support/survey-platform/survey-module/survey-flow/standard-elements/embedded-data/#SettingValuesFromTheSurveyURL)
```

2. (if applicable) use piped text to refer to those values in the question text

> 3. (optional) add a branch after the embedded data to have people skip the survey "if `id` is Equal to demo" 

Reviewer: ```suggestion
(Optional) Add a Branch after the Embedded Data block that checks whether 'id' equals 'demo'. If so, skip the survey. This can be useful for testing or demo participants.
```

Reviewer: Not sure if I edited that right
